### PR TITLE
fix #4339

### LINKED ISSUE #4339: PHP 8 - "There is no active transaction" when transaction is autocommitted
This is related to https://github.com/yiisoft/yii2/issues/18406 

As of PHP 8, PDO::commit() and rollBack() will throw a There is no active transaction PDOException if the transaction had been autocommitted, as MySQL will do whenever a schema change is made. Previously they would fail silently.

### What steps will reproduce the problem?
1. Connect to a MySQL database
2. Call Yii::app()->db->beginTransaction()
3. Make a schema change, such as add a new column to a table (the transaction will be autocommitted in the process)
4. Call Yii::app()->db->commit()

### What is the expected result?
PHP would move along to the next line of code, regardless of whether the app is running on PHP 5, 7, or 8.

### What do you get instead?
A PDOException is thrown with the message There is no active transaction when running PHP 8. 

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 1.1.24-dev
| PHP version      | 8.0.0
| Operating system | Centos 7

### PR DESCRIPTION
**Pull request by twisted1919**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #4339

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

File: framework/db/CDbTransaction.php [hunk 1]
```diff
@@ -61,7 +61,8 @@ public function commit()
 		if($this->_active && $this->_connection->getActive())
 		{
 			Yii::trace('Committing transaction','system.db.CDbTransaction');
-			$this->_connection->getPdoInstance()->commit();
+			if($this->_connection->getPdoInstance()->inTransaction())
```
Comment (marcovtwout) [inline framework/db/CDbTransaction.php:64]: @samdark Shouldn't the trace message be part of the if-block?

Comment (samdark) [reply framework/db/CDbTransaction.php:64]: Yes, I think so.

**Comment by twisted1919** [comment]
I can adjust this thing later this month if you wish and do another pull request, but for now, i simply don't have enough spare time to do it. 

If it's not acceptable, @marcovtwout can you please take care of it?

**Comment by marcovtwout** [comment]
@twisted1919 I've reviewed this change further and see that it is completely in line with the change in Yii 2 (https://github.com/yiisoft/yii2/pull/18407/files). The Yii 2 implementation also logs the debug/trace even if the transaction was already autocommitted, so I suggest we leave it as is.

**Comment by twisted1919** [comment]
Awesome :)

**Comment by samdark** [comment]
Fixed the tag. Thanks. @marcovtwout any ETA on release?

**Comment by marcovtwout** [comment]
@forevermatt I'm planning a regular release schedule, the next release should be no later then 2020-06-01. You can use master in the meantime.

**Comment by forevermatt** [comment]
That makes sense. Thanks!

**Pull request merged**